app/javascript/packs/puerto-rico-guidance.ts

```diff
@@ -0,0 +1,18 @@
+function showOrHidePuertoRicoGuidance(forStateCode) {
+  const puertoRicoGuidance = document.getElementById('puerto-rico-extra-text');
+  if (forStateCode === 'PR') {
+    puertoRicoGuidance?.classList?.remove('display-none');
+    puertoRicoGuidance?.classList?.add('display-block');
+  } else {
+    puertoRicoGuidance?.classList?.remove('display-block');
+    puertoRicoGuidance?.classList?.add('display-none');
+  }
+}
+
+function onStateSelectionChange() {
+  const stateSelector = <HTMLInputElement>document.getElementById('idv_form_state');
```

Comment:
I'm not familiar with this syntax, though surprisingly it seems to work!

More conventionally I might expect something looking like this, to assign the type of the variable:

```suggestion
  const stateSelector: HTMLInputElement = document.getElementById('idv_form_state');
```

Reply:
Doesn't work. getElementById returns an HTMLElement. The cast is necessary to turn it into an HTMLInputElement, with or without the variable type specified. I added the type.

Reply:
Changed.